tests/model_notebooks/test_notebooks.py

```diff
@@ -0,0 +1,361 @@
+        4. Tracks successes and failures
+        5. Logs summary of results
+        6. Raises AssertionError (exit code 1) if any notebooks failed
+
+        This ensures notebooks can fail without terminating the SLURM session.
+
+        Raises:
+            AssertionError: If any notebooks failed to execute successfully
+        """
+        logger.info("Starting notebook test run")
+        logger.info(f"Notebook directory: {self.notebook_directory}")
+        logger.info(f"Model directory: {self.model_dir}")
+        logger.info(f"Kernel name: {self.kernel_name}")
+        # Discover notebooks
+        self.notebooks_found = self._discover_notebook_paths()
+
+        # Handle empty directory case
+        if not self.notebooks_found:
+            logger.info("No notebooks found")
+            return
+
+        # Execute each notebook
+        for notebook_path in self.notebooks_found:
+            success = self._execute_notebook(notebook_path)
+            if success:
+                self.successful_notebooks.append(notebook_path)
+
+        # Log summary
+        total_count = len(self.notebooks_found)
+        success_count = len(self.successful_notebooks)
+        fail_count = len(self.failed_notebooks)
+
+        logger.info("=" * 60)
+        logger.info("Test run complete")
+        logger.info(f"{success_count}/{total_count} notebook(s) passed")
+
+        if self.failed_notebooks:
+            failed_names = [nb.name for nb in self.failed_notebooks.keys()]
+            logger.error(f"{fail_count} notebook(s) failed: {', '.join(failed_names)}")
+
+            # Raise assertion error to fail the test
+            raise AssertionError(
+                f"{fail_count} notebook(s) failed to execute: {', '.join(failed_names)}"
+            )
+        else:
+            logger.success("All notebooks passed!")
+
+    def get_summary(self) -> Dict[str, Any]:
+        """
+        Get a summary of the test run.
+
+        Returns:
+            Dictionary containing test run metadata:
+                - total_notebooks: Total number of notebooks found
+                - successful_notebooks: Number of successful executions
+                - failed_notebooks: Number of failed executions
+                - failed_notebook_names: List of failed notebook names
+                - notebook_directory: Path to notebook directory
+                - results_directory: Path to results directory
+        """
+        return {
+            "total_notebooks": len(self.notebooks_found),
+            "successful_notebooks": len(self.successful_notebooks),
+            "failed_notebooks": len(self.failed_notebooks),
+            "failed_notebook_names": [nb.name for nb in self.failed_notebooks.keys()],
+            "notebook_directory": str(self.notebook_directory),
+            "model_directory": str(self.model_dir),
+        }
+
+
+# Pytest test functions
+@pytest.mark.slow
+def test_interactive_notebooks() -> None:
```

Comment:
Nit: since we are running these non-interactively, can we think of a better name for this? Maybe we should call them "interactive context notebooks"?

Reply:
Yea Ill make the name more descriptive.